connectors/sources/confluence.py

```diff
@@ -993,46 +1005,55 @@ async def _page_blog_coro(self, api_query, target_type):
+                # but you can restrict either viewing or editing to certain users or groups.
+                if self.confluence_client.data_source_type == CONFLUENCE_CLOUD:
                     access_control = list(
-                        self._get_access_control_from_permission(
-                            permissions=permissions, target_type=target_type
-                        )
-                    )
-            else:
-                access_control = list(
-                    self._extract_identities_for_datacenter(response=restrictions)
-                )
-                if len(access_control) == 0:
-                    permission = await self.fetch_server_space_permission(
-                        space_key=space_key
+                        self._extract_identities(response=restrictions)
                     )
+                    if len(access_control) == 0:
+                        # Every space has its own independent set of permissions, managed by the space admin(s),
+                        # which determine the access settings for different users and groups.
+                        access_control = list(
+                            self._get_access_control_from_permission(
+                                permissions=permissions, target_type=target_type
+                            )
+                        )
+                else:
                     access_control = list(
-                        self.get_permission(
-                            permission=permission.get("permissions", {}).get(
-                                "VIEWSPACE", {}
+                        self._extract_identities_for_datacenter(response=restrictions)
+                    )
+                    if len(access_control) == 0:
+                        permission = await self.fetch_server_space_permission(
+                            space_key=space_key
+                        )
+                        access_control = list(
+                            self.get_permission(
+                                permission=permission.get("permissions", {}).get(
+                                    "VIEWSPACE", {}
+                                )
                             )
                         )
+                document = self._decorate_with_access_control(
+                    document=document, access_control=access_control
+                )
+                await self.queue.put((document, None))  # pyright: ignore
+                if attachment_count > 0:
+                    await self.fetchers.put(
+                        partial(self._attachment_coro, copy(document), access_control)
                     )
-            document = self._decorate_with_access_control(
-                document=document, access_control=access_control
+                    self.fetcher_count += 1
+        except Exception as exception:
+            self._logger.exception(
+                f"Error while fetching pages and blogposts: {exception}"
```

Comment:
```suggestion
                f"Error while fetching pages and blogposts with query '{api_query}': {exception}"
```

adding more info so that we can better reproduce if there's an issue